pkg/streamer/file.go

```diff
@@ -186,141 +187,90 @@ func fileSizeUpdated(path string, latestSize int64) (int, error) {
 	if err != nil {
 		return 0, terror.ErrGetRelayLogStat.Delegate(err, path)
 	}
-	currSize := fi.Size()
+	curSize := fi.Size()
 	switch {
-	case currSize == latestSize:
+	case curSize == latestSize:
 		return 0, nil
-	case currSize > latestSize:
-		log.L().Debug("size of relay log file has been changed", zap.String("file", path), zap.Int64("old size", latestSize), zap.Int64("size", currSize))
+	case curSize > latestSize:
+		log.L().Debug("size of relay log file has been changed", zap.String("file", path),
+			zap.Int64("old size", latestSize), zap.Int64("size", curSize))
 		return 1, nil
 	default:
-		log.L().Error("size of relay log file has been changed", zap.String("file", path), zap.Int64("old size", latestSize), zap.Int64("size", currSize))
+		log.L().Error("size of relay log file has been changed", zap.String("file", path),
+			zap.Int64("old size", latestSize), zap.Int64("size", curSize))
 		return -1, nil
 	}
 }
 
-// relaySubDirUpdated checks whether the relay sub directory updated
-// including file changed, created, removed, etc.
-func relaySubDirUpdated(ctx context.Context, watcherInterval time.Duration, dir string,
+// relayLogUpdatedOrNewCreated checks whether current relay log file is updated or new relay log is created.
+// we check the size of the file first, if the size is the same as the latest file, we assume there is no new write
+// so we need to check relay meta file to see if the new relay log is created.
+// this func will be blocked until current filesize changed or meta file updated or context cancelled.
```

Comment:
Should we watch the `server-uuid.index` file? For master-slave swith

Reply:
I think the function `needSwitchSubDir` is used to handle master-slave switching? there is another goroutine in `parseFile` that already does this